sessionctx/binloginfo/binloginfo.go

```diff
@@ -46,11 +49,21 @@ func GetPrewriteValue(ctx context.Context, createIfNotExists bool) *binlog.Prewr
 func WriteBinlog(bin *binlog.Binlog, clusterID uint64) error {
 	commitData, _ := bin.Marshal()
 	req := &binlog.WriteBinlogReq{ClusterID: clusterID, Payload: commitData}
-	resp, err := PumpClient.WriteBinlog(goctx.Background(), req)
-	if err == nil && resp.Errmsg != "" {
-		err = errors.New(resp.Errmsg)
+
+	// Retry many times because we may raise CRITICAL error here.
+	var err error
+	for i := 0; i < 20; i++ {
+		var resp *binlog.WriteBinlogResp
+		resp, err = PumpClient.WriteBinlog(goctx.Background(), req)
+		if err == nil && resp.Errmsg != "" {
+			err = errors.New(resp.Errmsg)
```

Comment:
log the error here, that information is important, and it may be overwrite by the next loop.